### PR TITLE
feature(frontend): adding synthetic attribute flag to FE instrumentation

### PR DESCRIPTION
**Pull request by xoscar**
# Changes

This PR includes a new span attribute coming from the FE instrumentation when triggering a fetch request to the API endpoints.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs](../docs/) folder

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

### CODE REVIEW

File: src/frontend/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ RUN npm run build
 
 FROM node:18-alpine AS runner
 WORKDIR /app
-RUN apk add --no-cache protoc
+RUN apk add --no-cache protobuf-dev protoc
```
Comment (fatsheep9146) [inline src/frontend/Dockerfile:21]: I think this change is duplicated in #631, #632 #633. I'm not sure, is it necessary to add this change to every pr?

File: src/frontend/utils/telemetry/Instrumentation.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+const { getNodeAutoInstrumentations } = require('@opentelemetry/auto-instrumentations-node');
+const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-grpc');
 
 const sdk = new opentelemetry.NodeSDK({
   traceExporter: new OTLPTraceExporter(),
-  instrumentations: [ getNodeAutoInstrumentations() ]
-})
+  instrumentations: [
+    getNodeAutoInstrumentations({
+      '@opentelemetry/instrumentation-fs': {
```
Comment (puckpuck) [inline src/frontend/utils/telemetry/Instrumentation.js:9]: why are we disabling this auto-instrumentation?

**Review by puckpuck — CHANGES_REQUESTED** [review]

File: src/frontend/utils/telemetry/Instrumentation.js (hunk 2 above)
Comment (puckpuck) [reply src/frontend/utils/telemetry/Instrumentation.js:9]: We remove this from this PR, and leave it in #632 which is requires the SDK version upgrade.

File: src/frontend/utils/telemetry/FrontendTracer.ts [hunk 3]
```diff
@@ -43,6 +43,9 @@ const FrontendTracer = async (collectorString: string) => {
         '@opentelemetry/instrumentation-fetch': {
           propagateTraceHeaderCorsUrls: /.*/,
           clearTimingResources: true,
+          applyCustomAttributesOnSpan(span) {
```
Comment (puckpuck) [inline src/frontend/utils/telemetry/FrontendTracer.ts:46]: This unique way to add attributes on spans at the instrumentation initialization needs to be documented in the [frontend docs for browser instrumentation](https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/services/frontend.md#browser-instrumentation)

**Review by puckpuck — CHANGES_REQUESTED** [review]

File: src/frontend/Dockerfile (hunk 1 above)
Comment (xoscar) [reply src/frontend/Dockerfile]: This has been removed 😄

File: src/frontend/utils/telemetry/Instrumentation.js (hunk 2 above)
Comment (xoscar) [reply src/frontend/utils/telemetry/Instrumentation.js]: Done 😄

File: src/frontend/utils/telemetry/FrontendTracer.ts (hunk 3 above)
Comment (xoscar) [reply src/frontend/utils/telemetry/FrontendTracer.ts:46]: Done 😄

**Review by puckpuck — APPROVED** [review]

**Comment by puckpuck** [comment]
@xoscar I know it's the holidays so you may not get this, but when you get a chance please update the branch so we can get this one merged.

In the future, if you do a PR from your personal fork instead of an organizational one, we can update the branch for you.

**Comment by xoscar** [comment]
@puckpuck makes sense! I'll start doing that, thanks for the feedback! 😄

**Pull request merged**